app/ydoc-shared/src/ast/parse.ts

```diff
@@ -572,6 +565,11 @@ export function printAst(
     if (!isTokenId(child.node) && ast.module.get(child.node) === undefined) continue
     if (prevIsNewline) currentLineIndent = child.whitespace
     const token = isTokenId(child.node) ? ast.module.getToken(child.node) : undefined
+    // Every line in a block starts with a newline token. In an AST produced by the parser, the newline token at the
+    // first line of a module is zero-length. In order to handle whitespace correctly if the lines of a module are
+    // rearranged, if a zero-length newline is encountered within a block, it is printed as an ordinary newline
+    // character, and if an ordinary newline is found at the beginning of the output, it is not printed; however if the
+    // output begins with a newline including a (plain) comment, we print the line as we would in any other block.
```

Comment:
> however if the output begins with a newline including a (plain) comment, we print the line as we would in any other block.

I cannot find any code handling the mentioned case in a specific way.

Reply:
It's the newline token content check below; plain comments are included in newline tokens.